plugin-infra/go-plugin-domain/src/main/java/com/thoughtworks/go/plugin/domain/configrepo/Capabilities.java

```diff
@@ -46,17 +50,35 @@ public void setSupportsParseContent(boolean supportsParseContent) {
         this.supportsParseContent = supportsParseContent;
     }
 
+    public boolean isSupportsListConfigFiles() {
+        return supportsListConfigFiles;
+    }
+
+    public void setSupportsListConfigFiles(boolean supportsListConfigFiles) {
+        this.supportsListConfigFiles = supportsListConfigFiles;
+    }
+
+    public boolean isSupportsUserDefinedProperties() {
+        return supportsUserDefinedProperties;
+    }
+
+    public void setSupportsUserDefinedProperties(boolean supportsUserDefinedProperties) {
+        this.supportsUserDefinedProperties = supportsUserDefinedProperties;
+    }
+
```

Comment:
should we consider using Lombok?
```java
@Getter
@Setter
@EqualsAndHashCode
@Accessors(chain = true)
@NoArgsConstructor(access = AccessLevel.PUBLIC)
@AllArgsConstructor(access = AccessLevel.NONE)
public class Capabilities2 {
    private boolean supportsPipelineExport;
    private boolean supportsParseContent;
    private boolean supportsListConfigFiles;
    private boolean supportsUserDefinedProperties;
}
```

Reply:
@GaneshSPatil I thought about that too. Hesitated to mess with the gradle deps at the time. But, yes, we probably should. All these objects are just `@Data` essentially.